### DISCUSSION TITLE
import "Now Playing" (aka CUE) information from CUE files directly into existing Media Items

### DISCUSSION THREAD

**Discussion by t0mtaylor** [AzuraCast/AzuraCast#7260] [Ideas]
Now Playing information: import "Now Playing" (aka CUE) information from CUE files directly into existing Media Items, with support for cuegenerator.net cue files

For example, a Radio presenter has provided a 2 hour recording of their radio show, and there is an mp3 file of the recording, and a cue file with all the breakpoints and tracknames - it would be great in Azuracast or AutoCue imports this cue file and displays the metadata at the correct time as per the cue files cue points for now playing information

It is opposite to this idea which is to record the data from a live recording into a cue file, and both of these ideas should be achievable together: https://github.com/AzuraCast/AzuraCast/issues/6872

More details on how some audio software can generator cue points for the cue files: https://cuegenerator.net/help and on github https://github.com/DmitryVarennikov/cuegenerator-svelte

Performer and title track separators:

' - ' — 45 (hyphen-minus)
' – ' — 8211 (en dash)
' ‒ ' — 8210 (figure dash)
' — ' — 8212 (em dash)
' ― ' — 8213 (horizontal bar)

Timings recognition:

[08:45] 03. 8 Ball → 08:45
01.[18:02] Giuseppe → 18:02
10:57 02. Space Manoeuvres → 10:57
56:53 T.O.M' → 56:53
1:02:28 Mossy → 62:28
Regions list recognition
Sony Sound Forge format dd:dd:dd[.,]dd
Adobe Audition format dd:dd:dd:dd
Audacity format ddddd.dddddd

@Moonbase59

Live365 have good documentation on cue and csv file support - https://help.live365.com/support/solutions/articles/43000697897-multitrack-cue-sheet-and-csv-file-requirements

**Comment by Moonbase59** [comment]
Sorry, Autocue will never support this. It looks like both need to use `liq_cue_in` and `liq_cue_out`, and we’d have to analyse and store multiple data points for what is actually just _one_ file.

As a file format, it’s also not easy to handle: You’d have to seek around in a large file and _keep multiple parts of it ready_ for transitions and the like. The cue points given are also often not exact enough, and some file types (like MP3 which many would possibly use) are notably hard or impossible to seek exactly.

Conversion to another file type, like cut-up WAV pieces _before_ we do any autocueing seems too high a burden on a typical small station VM. It would also suffer from above-said problems.

_Attaching_ a cue sheet to large files like show recordings, pre-recorded shows, DJ sets and podcasts _makes sense_ to approximately know what was happening when, and for keeping "now playing" information. You _can_ autocue such files, but it will only ever affect the start and end of the _complete file_, to ensure a smooth transition.

---

That said, I believe that Liquidsoap, the underlying streaming software, will eventually come up (or can already?) read standard cue sheet information, so _playout_ with the correct artist/title/etc. should be possible one day.

**Reply by t0mtaylor** [reply]
@Moonbase59 yea I was thinking more along the lines of just reading the cue sheet, which is kb in size - this way importing this data into the media item could be the database and not the media file

for example some options:

1. running a thread/cron job to read the cue sheet (php/nodejs/python/bash?) and update the now playing information as the show runs in the playlist for that hour or however long the track and cue sheet is
2. convert to json and just load the latest song title based on time less than the next one in the list

This way, there is no chopping up of crazy large files (who is using wav, they should be using flac!) as your just reading the metadata from the cue file

Some shows have say 3x 20min parts each with its own cue sheet - I can provide examples if required or we can create some

I would avoid processing with liquidsoap unless we have to!

**Comment by t0mtaylor** [comment]
Also noted that this is something @Moonbase59 would like, which would record the cue files, but we also need to ingest and read the cue files to display the meta data

I don't think its a liquidsoap issue directly, more azuracast needs to write the files and read/import them into the db when they exist with an mp3 of a show which is added to azuracast

https://features.azuracast.com/suggestions/518478/add-cue-file-to-live-recordings

https://github.com/savonet/liquidsoap/issues/3608

This looks promising https://github.com/AzuraCast/AzuraCast/issues/6872
